include/pybind11/detail/class.h

```diff
@@ -602,9 +602,9 @@ extern "C" inline int pybind11_getbuffer(PyObject *obj, Py_buffer *view, int fla
         return -1;
     }
     view->obj = obj;
-    view->ndim = 1;
     view->internal = info;
     view->buf = info->ptr;
+    view->ndim = (int) info->ndim;
```

Comment:
Adding the default with a link to this PR would be ideal so that it is clear why we chose to do this.
```
view->ndim = 1; // See PR #5407
```